domain/secret/state/state.go

```diff
@@ -1045,6 +1052,83 @@ func (st State) GetSecret(ctx context.Context, uri *coresecrets.URI) (*coresecre
 	return secrets[0], nil
 }
 
+// GetRotationExpiryInfo returns the rotation expiry information for the specified secret.
+func (st State) GetRotationExpiryInfo(ctx context.Context, uri *coresecrets.URI) (*secret.RotationExpiryInfo, error) {
+	db, err := st.DB()
+	if err != nil {
+		return nil, errors.Trace(err)
+	}
+
+	stmt, err := st.Prepare(`
+WITH rev AS
+    (SELECT uuid,MAX(revision) AS latest_revision
+    FROM    secret_revision
+    WHERE   secret_id = $secretID.id)
+SELECT sp.policy AS &secretInfo.policy,
+       sro.next_rotation_time AS &secretInfo.next_rotation_time,
+       sre.expire_time AS &secretInfo.latest_expire_time,
+       rev.latest_revision AS &secretInfo.latest_revision
+FROM secret_metadata sm, rev
+INNER JOIN secret_rotate_policy sp ON sp.id = sm.rotate_policy_id
+LEFT JOIN secret_rotation sro ON sro.secret_id = sm.secret_id
+LEFT JOIN secret_revision_expire sre ON sre.revision_uuid = rev.uuid
+WHERE sm.secret_id = $secretID.id`, secretID{}, secretInfo{})
+
+	if err != nil {
+		return nil, errors.Trace(err)
+	}
+
+	var result secretInfo
+	if err := db.Txn(ctx, func(ctx context.Context, tx *sqlair.TX) error {
+		err := tx.Query(ctx, stmt, secretID{ID: uri.ID}).Get(&result)
+		if errors.Is(err, sqlair.ErrNoRows) {
+			return fmt.Errorf("secret %q not found%w", uri, errors.Hide(secreterrors.SecretNotFound))
+		}
+		return errors.Trace(err)
+	}); err != nil {
+		return nil, errors.Trace(domain.CoerceError(err))
+	}
+	info := &secret.RotationExpiryInfo{
+		RotatePolicy:   coresecrets.RotatePolicy(result.RotatePolicy),
+		LatestRevision: result.LatestRevision,
+	}
+	if !result.NextRotateTime.IsZero() {
```

Comment:
Should we add a sql type like we did with string

Reply:
What do you mean?